hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieCompactedLogRecordScanner.java

```diff
@@ -72,142 +78,148 @@
-              log.info("Found a corrupt block in " + logFile.getPath());
-              // If there is a corrupt block - we will assume that this was the next data block
-              lastBlocks.push(r);
-              break;
-          }
-        }
 
-      } catch (IOException e) {
-        throw new HoodieIOException("IOException when reading log file " + logFile);
-      }
-      // merge the last read block when all the blocks are done reading
-      if(!lastBlocks.isEmpty()) {
-        log.info("Merging the final data blocks in " + logFile.getPath());
-        merge(records, lastBlocks);
+          } catch (IOException e) {
+              throw new HoodieIOException("IOException when reading log file " + logFile);
+          }
+          // merge the last read block when all the blocks are done reading
+          if(!lastBlocks.isEmpty()) {
+              log.info("Merging the final data blocks in " + logFile.getPath());
+              merge(records, lastBlocks);
+          }
       }
-    }
-    this.logRecords = Collections.unmodifiableCollection(records.values());
-    this.totalRecordsToUpdate = records.size();
+      this.logRecords = Collections.unmodifiableCollection(records.values());
+      this.totalRecordsToUpdate = records.size();
   }
 
-  /**
-   * Iterate over the GenericRecord in the block, read the hoodie key and partition path
-   * and merge with the HoodieAvroPayload if the same key was found before
-   *
-   * @param dataBlock
-   */
-  private Map<String, HoodieRecord<HoodieAvroPayload>> loadRecordsFromBlock(HoodieAvroDataBlock dataBlock) {
-    Map<String, HoodieRecord<HoodieAvroPayload>> recordsFromLastBlock = Maps.newHashMap();
-    List<IndexedRecord> recs = dataBlock.getRecords();
-    totalLogRecords.addAndGet(recs.size());
-    recs.forEach(rec -> {
-      String key = ((GenericRecord) rec).get(HoodieRecord.RECORD_KEY_METADATA_FIELD)
-        .toString();
-      String partitionPath =
-          ((GenericRecord) rec).get(HoodieRecord.PARTITION_PATH_METADATA_FIELD)
-              .toString();
-      HoodieRecord<HoodieAvroPayload> hoodieRecord = new HoodieRecord<>(
-          new HoodieKey(key, partitionPath),
-          new HoodieAvroPayload(Optional.of(((GenericRecord) rec))));
-      if (recordsFromLastBlock.containsKey(key)) {
-        // Merge and store the merged record
-        HoodieAvroPayload combinedValue = recordsFromLastBlock.get(key).getData()
-          .preCombine(hoodieRecord.getData());
-        recordsFromLastBlock
-          .put(key, new HoodieRecord<>(new HoodieKey(key, hoodieRecord.getPartitionPath()),
-                        combinedValue));
-      } else {
-        // Put the record as is
-        recordsFromLastBlock.put(key, hoodieRecord);
-      }
-    });
-    return recordsFromLastBlock;
-  }
+    /**
+     * Iterate over the GenericRecord in the block, read the hoodie key and partition path
+     * and merge with the HoodieAvroPayload if the same key was found before
```

Comment:
change refrence to "HoodieAvroPayload" in comments..

Reply:
and also probably some explanation on - its sufficient to just merge the log records, since base data is merged on previous compaction.